newIDE/app/src/Utils/GDevelopServices/Project.js

```diff
@@ -0,0 +1,271 @@
+  authenticatedUser: AuthenticatedUser,
+  cloudProjectId: string,
+  apiCall: () => Promise<T>
+): Promise<T> => {
+  try {
+    const response = await apiCall();
+    return response;
+  } catch (error) {
+    if (error.response && error.response.status === 403) {
+      await getCredentialsForCloudProject(authenticatedUser, cloudProjectId);
+      const response = await apiCall();
+      return response;
+    }
+    throw error;
+  }
+};
+
+const getVersionIdFromPath = (path: string): string => {
+  let cleanedPath = path;
+  const searchParamsStartIndex = path.indexOf('?');
+  if (searchParamsStartIndex >= 0) {
+    cleanedPath = path.substring(0, searchParamsStartIndex);
+  }
+  const filenameStartIndex = cleanedPath.lastIndexOf('/') + 1;
+  const filenameEndIndex = cleanedPath.indexOf('.zip');
+  return path.substring(filenameStartIndex, filenameEndIndex);
+};
+
+export const getCredentialsForCloudProject = async (
+  authenticatedUser: AuthenticatedUser,
+  cloudProjectId: string
+): Promise<boolean> => {
+  const { getAuthorizationHeader, firebaseUser } = authenticatedUser;
+  if (!firebaseUser) return false;
+
+  const { uid: userId } = firebaseUser;
+  const authorizationHeader = await getAuthorizationHeader();
+  const response = await axios.get(
+    `${GDevelopProjectApi.baseUrl}/project/${cloudProjectId}/action/authorize`,
+    {
+      headers: { Authorization: authorizationHeader },
+      params: { userId },
+      validateStatus: status => true,
+      withCredentials: true, // Necessary to save cookie returned by the server.
+    }
+  );
+  return response.status >= 200 && response.status < 400;
+};
+
+export const clearCloudProjectCredentials = async (): Promise<void> => {
+  await axios.get(`${GDevelopProjectApi.baseUrl}/action/clear-authorization`, {
+    withCredentials: true, // Necessary to save cookie returned by the server.
+  });
+};
+
+export const createCloudProject = async (
+  authenticatedUser: AuthenticatedUser,
+  cloudProjectCreationPayload: { name: string }
+): Promise<?CloudProject> => {
+  const { getAuthorizationHeader, firebaseUser } = authenticatedUser;
+  if (!firebaseUser) return null;
+
+  const { uid: userId } = firebaseUser;
+  const authorizationHeader = await getAuthorizationHeader();
+  const response = await axios.post(
+    `${GDevelopProjectApi.baseUrl}/project`,
+    cloudProjectCreationPayload,
+    {
+      headers: { Authorization: authorizationHeader },
+      params: { userId },
+    }
+  );
+  return response.data;
+};
+
+export const commitVersion = async (
+  authenticatedUser: AuthenticatedUser,
+  cloudProjectId: string,
+  zippedProject: any
+): Promise<?string> => {
+  const { getAuthorizationHeader, firebaseUser } = authenticatedUser;
+  if (!firebaseUser) return;
+
+  const { uid: userId } = firebaseUser;
+  const authorizationHeader = await getAuthorizationHeader();
+  // get presigned url to upload new version (that contains new version id)
```

Comment:
Ubernit: capitals + periods in comments.
```suggestion
  // Get a presigned url to upload a new version (the URL will contain the new version id).
```